tests/ondemand/ondemand_basictests.cpp

```diff
@@ -181,47 +180,66 @@ namespace number_tests {
   // adapted by D. Lemire
   inline uint64_t f64_ulp_dist(double a, double b) {
     uint64_t ua, ub;
-    memcpy(&ua, &a, sizeof(ua));
-    memcpy(&ub, &b, sizeof(ub));
-    if ((int64_t)(ub ^ ua) >= 0)
-      return (int64_t)(ua - ub) >= 0 ? (ua - ub) : (ub - ua);
+    std::memcpy(&ua, &a, sizeof(ua));
```

Comment:
I believe there is a copy of this function in tests/basictests.cpp, yes? I didn't think I'd modified it, so the same change might need to be made there.

Reply:
Ok, maybe it should be pulled out to a separate file in that case?

Reply:
Found three copies, extracted them to a separate file and added a unit test for it. But I think it looks suspicious that the same constant 0x800... is used for both the float and double version?